services/wallet/router/router.go

```diff
@@ -213,7 +230,7 @@ func (r *Router) SuggestedRoutes(ctx context.Context, input *requests.RouteInput
 		r.activeRoutesMutex.Lock()
 		r.activeRoutes = suggestedRoutes
 		r.activeRoutesMutex.Unlock()
-		if suggestedRoutes != nil && err == nil {
+		if suggestedRoutes != nil && err == nil && !r.isRouteCanceled() {
 			// subscribe for updates
 			for _, path := range suggestedRoutes.Best {
 				err = r.subscribeForUdates(path.FromChain.ChainID)
```

Comment:
you could potentially have routeCanceled at false at the time of executing the if, and canceling when starting to iterate. you need to acquire the lock until subscribe for update is done

Reply:
@alaibe yes, I see your point, I was relaying that those two calls cannot be done async, but now updated to a safer version. Thanks.